packages/oas/src/converter/DefaultConverter.ts

```diff
@@ -643,45 +689,83 @@ export class DefaultConverter implements Converter {
       preferredUrls = urls;
     }
 
-    return sample({
+    return this.sample({
       type: 'array',
       examples: preferredUrls
     });
   }
 
   private parseUrls(spec: OpenAPI.Document): string[] {
     if (this.isOASV3(spec) && spec.servers?.length) {
-      return spec.servers.map((server: OpenAPIV3.ServerObject) => {
-        const variables = server.variables || {};
-        const templateUrl = template.parse(server.url);
-        const params = {};
-
-        for (const [param, variable] of Object.entries(variables)) {
-          const data = sample(variable, {}, spec);
-          Object.assign(params, { [param]: data });
-        }
-
-        return removeTrailingSlash(templateUrl.expand(params));
-      });
+      return this.parseServers(spec);
     }
 
     if (this.isOASV2(spec) && spec.host) {
-      const basePath: string =
-        typeof spec.basePath !== 'undefined'
-          ? removeLeadingSlash(spec.basePath)
-          : '';
-      const host: string = removeTrailingSlash(spec.host);
-      const schemes: string[] =
-        typeof spec.schemes !== 'undefined' ? spec.schemes : ['https'];
-
-      return schemes.map(
-        (x: string) => `${x}://${removeTrailingSlash(`${host}/${basePath}`)}`
-      );
+      return this.parseHost(spec);
     }
 
     return [];
   }
 
+  private parseHost(spec: OpenAPIV2.Document): string[] {
+    const basePath: string =
+      typeof spec.basePath !== 'undefined'
+        ? removeLeadingSlash(spec.basePath)
+        : '';
+    const host: string = removeTrailingSlash(spec.host);
+    const schemes: string[] =
+      typeof spec.schemes !== 'undefined' ? spec.schemes : ['https'];
+
+    return schemes.map(
+      (x: string) => `${x}://${removeTrailingSlash(`${host}/${basePath}`)}`
+    );
+  }
+
+  private parseServers(spec: OpenAPIV3.Document): string[] {
+    return spec.servers.map((server: OpenAPIV3.ServerObject, idx: number) => {
+      const variables = server.variables || {};
+      const params = Object.entries(variables).reduce(
+        (acc, [param, variable]: [string, OpenAPIV3.ServerVariableObject]) => {
+          const jsonPointer = pointer.compile([
+            'servers',
+            idx.toString(),
+            'variables',
+            param
+          ]);
+          const data = this.sample(
+            variable,
+            {},
+            {
+              spec,
+              jsonPointer
+            }
+          );
+
+          return { ...acc, [param]: data };
+        },
+        {}
+      );
+      const templateUrl = template.parse(server.url);
+
+      return removeTrailingSlash(templateUrl.expand(params));
+    });
+  }
+
+  private sample(
+    schema: Schema,
+    options?: Options,
+    context?: {
+      spec?: OpenAPI.Document;
+      jsonPointer?: string;
+    } & Options
```

Comment:
no need in `Options` as part of context?
```suggestion
    }
```